### PR TITLE
Added error message, and added check for empty strings in parse_internal

### LINKED ISSUE #18070: Very bad error message for trying to load an empty string as an asset.
## What problem does this solve or what need does it fill?

This is the error message, "" is not printed and asset type is missing, making it completely unusable for debugging purpose.

```
Encountered an I/O error while loading asset: Is a directory (os error 21)
```

## What solution would you like?

Change this somehow.

### PR DESCRIPTION
**Pull request by DylanD42**
# Objective

- Prevents from empty strings when loading assets
- Fixes #18070

## Solution

- Added a check into parse_internal that checks if asset_path is empty, if so throw a EmptyPath panic, standard to other parsing

## Testing

- Added a test that shows a example of asset loading with a correct and a incorrect path, as intended. Also added a test for empty string that will result in a panic.
- Further testing on varying file types may be beneficial.
#

### CODE REVIEW

File: tests/asset/empty_asset.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+use bevy::prelude::*;
+use bevy::asset::LoadState;
+
+#[test]
+fn test_asset_load_valid_file() {
```
Comment (andriyDev) [inline tests/asset/empty_asset.rs:5]: `bevy_asset/src/lib.rs` already contains a bunch of tests, which you could use as examples. It might be better to just add a new test there rather than start something new here (especially since those tests already have a bunch of utilities for writing tests more conveniently).
